### PR TITLE
gcs: fix regression with input wizard "freezing"

### LINKED ISSUE #407: REGRESSION: I broke the input wizard
I believe it may be an interaction with the ensuring-valid-values logic on the flight side.

### PR DESCRIPTION
**Pull request by mlyle**
Fixes #407

[Review on Reviewable](https://reviewable.io/reviews/d-ronin/dronin/416)

### CODE REVIEW

**Comment by mlyle** [comment]
I still need to test this.  Just staging this here.

**Comment by mlyle** [comment]
Jenkins build artifacts please.

File: ground/gcs/src/plugins/config/configinputwidget.cpp [hunk 1]
```diff
@@ -595,12 +600,11 @@ void ConfigInputWidget::wizardSetUpStep(enum wizardSteps step)
         {
             // Preserve the inverted status
             if(manualSettingsData.ChannelMin[i] <= manualSettingsData.ChannelMax[i]) {
-                manualSettingsData.ChannelMin[i]=manualSettingsData.ChannelNeutral[i];
-                manualSettingsData.ChannelMax[i]=manualSettingsData.ChannelNeutral[i];
+                manualSettingsData.ChannelMin[i]=manualSettingsData.ChannelNeutral[i] - INITIAL_OFFSET;
```
Comment (tracernz) [inline ground/gcs/src/plugins/config/configinputwidget.cpp]: Spacing around equals in these lines.

**Comment by mlyle** [comment]
jenkins build artifacts please

**Comment by pug398** [comment]
Tested and flown with inverted sbus. LGTM.

**Comment by mlyle** [comment]
@pug398 Cool, thanks.  :)   I'm going to try and play with it a little and then hopefully this can speedily merge.

**Comment by jihlein** [comment]
Successfully tested on SciSky Naze32, no configuration wizard issues.

Successfully tested Sparky with D4R-II PPM/FrSky Senor Hub, no configuration wizard issues,

GCS running on Win7 Laptop.

**Comment by mlyle** [comment]
Tested all the edge cases, etc.  It does reasonably.  Doesn't manifest the bug.

**Comment by dustin** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[ground/gcs/src/plugins/config/configinputwidget.cpp, line 1022 \[r2\]](https://reviewable.io:443/reviews/d-ronin/dronin/416#-K7eH8bU9BnfnRevxmiQ:-K7eH8bU9BnfnRevxmiR:-635496646) ([raw file](https://github.com/d-ronin/dronin/blob/0e4031d2ed96857da6f699349647fd6b3678a34f/ground/gcs/src/plugins/config/configinputwidget.cpp#L1022)):*
I don't get the default here.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/d-ronin/dronin/416)*

**Comment by mlyle** [comment]
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[ground/gcs/src/plugins/config/configinputwidget.cpp, line 1022 \[r2\]](https://reviewable.io:443/reviews/d-ronin/dronin/416#-K7eH8bU9BnfnRevxmiQ:-K7eHmcPAGmNlfONntpQ:1227148622) ([raw file](https://github.com/d-ronin/dronin/blob/0e4031d2ed96857da6f699349647fd6b3678a34f/ground/gcs/src/plugins/config/configinputwidget.cpp#L1022)):*
There was special purpose code to "center" the throttle channel properly.  I added it for switches.  Rather than an if/else tree, I handled those 3 cases (throttle, arming, flight mode).

Compilers often warn if you switch on an enum and don't handle all potential values, so I added a default.  break breaks out of the switch statement, not the for loop.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/d-ronin/dronin/416)*

**Comment by dustin** [comment]
:lgtm:

---

Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/d-ronin/dronin/416#-:-K7eILPdBGCbctasEQNb:1417088721)*

**Pull request merged**